packages/ra-input-rich-text/src/index.js

```diff
@@ -45,18 +47,23 @@ export class RichTextInput extends Component {
     }
 
     onTextChange = () => {
-        this.props.input.onChange(this.editor.innerHTML);
+        const value =
+            this.editor.innerHTML == '<p><br></p>' ? '' : this.editor.innerHTML;
```

Comment:
Quill doesn't like void. Removing the content of a quill editor results in a non-empty content (!).